internal/pkg/comm/creds.go

```diff
@@ -28,26 +30,64 @@ var (
 // NewServerTransportCredentials returns a new initialized
 // grpc/credentials.TransportCredentials
 func NewServerTransportCredentials(
-	serverConfig *tls.Config,
+	serverConfig *TLSConfig,
 	logger *flogging.FabricLogger) credentials.TransportCredentials {
-
 	// NOTE: unlike the default grpc/credentials implementation, we do not
 	// clone the tls.Config which allows us to update it dynamically
-	serverConfig.NextProtos = alpnProtoStr
+	serverConfig.lock.Lock()
+	defer serverConfig.lock.Unlock()
+	serverConfig.config.NextProtos = alpnProtoStr
 	// override TLS version and ensure it is 1.2
-	serverConfig.MinVersion = tls.VersionTLS12
-	serverConfig.MaxVersion = tls.VersionTLS12
+	serverConfig.config.MinVersion = tls.VersionTLS12
+	serverConfig.config.MaxVersion = tls.VersionTLS12
 	return &serverCreds{
 		serverConfig: serverConfig,
 		logger:       logger}
 }
 
 // serverCreds is an implementation of grpc/credentials.TransportCredentials.
 type serverCreds struct {
-	serverConfig *tls.Config
+	serverConfig *TLSConfig
 	logger       *flogging.FabricLogger
 }
 
+type TLSConfig struct {
+	config *tls.Config
+	lock   *sync.Mutex
+}
+
+func NewTLSConfig(config *tls.Config, lock *sync.Mutex) *TLSConfig {
+	return &TLSConfig{
+		config: config,
+		lock:   lock,
+	}
+}
+
+func (t *TLSConfig) Config() tls.Config {
+	t.lock.Lock()
```

Comment:
Let's use a read lock here instead so you can do several TLS handshakes in parallel.